SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+   }
+}
+```
+An example of this could be:
+```json
+{
+   "honeyPrice": {
+      "TTL": 3600,
+       "price": {
+           "1566396921": 80000,
+           "1566400521": 80800
+       },
+       "currency": "Wei"
+   }
+}
+```
+where we see that the TTL is `3600` (1 hour), and that the oracle has quoted a total of two prices. If the current time is in between `1566396921` and `1566400521`, the `honeyToMoneyOracle` will return the price of `80000`, if the time is after `1566400521`, it will return the price of `80800`. Since the `TTL` is `3600`, `honeyToMoneyOracle`s are guaranteed to have queried the external oracle in between this time, thus ensuring that they all update simultaneously.
+* Any answer from the external oracle will be valid to the `honeyToMoneyOracle` for `TTL` seconds.
+* The applied price is chosen by looking up the price corresponding to the `validFrom` which is in the most recent past.
+* Upon receipt of payment or intent to send a payment, nodes will query the `honeyToMoneyOracle` who checks its local cache for the `honeyPrice` object. If it is not valid anymore or non-existent, the external oracle will be queried. Otherwise, the cached price which is in the most recent past will be returned.
+
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+### General rationale for choosing oracles
+There exists a need for the price of Swarm to change over time. However, due to the nature of Swarm, there is a need for this price to be decided network-wide, instead of peer-to-peer. Specifically:
+- In order for a node to have Kademlia connectivity, it must be connected to all peers in his most proximate bin and *at least* one peer for each proximity order up to the saturation depth (see [architecture docs](https://swarm-guide.readthedocs.io/en/latest/architecture.html#overlay-network)). For SWAP to work, nodes must have the certainty that they will be able to reach an agreement on the price when they have to settle their balance. Hence, in the situation that nodes agree on the price peer-to-peer, this price *must* be decided upon the point when nodes connect. As it might happen that they don't come to an agreement on price and hence don't connect, Kademlia connectivity cannot be guaranteed when prices are to be decided peer-to-peer—in particular because of the requirement of connectivity to all nodes in the most proximate bin.
+- Zero marginal costs of serving data with perfect competition means that when nodes would engage in peer-to-peer price discovery, the price will tend to zero and a sub-optimal equilibrium of near-zero prices will be reached.
```

Comment:
Hayek and Friedman would roll in their graves. 

This is a _very_ bold claim, and I would like to see this backed up with theory.